git-changelog: add page (#7359)
tldr-pages/tldr: pages/common/git-changelog.md

# git changelog

> Generate a changelog report from repository commits and tags.
> Part of `git-extras`.
> More information: <https://github.com/tj/git-extras/blob/master/Commands.md#git-changelog>.

> - Update existing file or creating a new History.md file with pretty formatted output:

Reviewer: What is a non `pretty` output?
```suggestion
- Update existing file or create a new `History.md` file with the commit messages since the latest Git tag:
```

Author: @mfrw can you take a look at this and give me your thoughts as well?

`git changelog`

- List commits from the current version:

`git changelog --list`

> - List a range of commits from 2.1.0 to now:

Reviewer: ```suggestion
- List a range of commits from the tag named `2.1.0` to now:
```

> `git changelog --list --start-tag 2.1.0`

Reviewer 2: ```suggestion
`git changelog --list --start-tag {{2.1.0}}`
```

> - List pretty formatted range of commits between 0.5.0 and 1.0.0:

Reviewer: ```suggestion
- List pretty formatted range of commits between the tag `0.5.0` and the tag `1.0.0`:
```

> `git changelog --start-tag 0.5.0 --final-tag 1.0.0`

Reviewer 2: ```suggestion
`git changelog --start-tag {{0.5.0}} --final-tag {{1.0.0}}`
```

> - List pretty formatted range of commits between 0b97430 and 1.0.0:

Reviewer: ```suggestion
- List pretty formatted range of commits between the commit `0b97430` and the tag `1.0.0`:
```

`git changelog --start-commit 0b97430 --final-tag 1.0.0`

> - Specify a file for output:

Reviewer: ```suggestion
- Specify `CHANGELOG.md` as the output file:
```

> `git changelog --start-commit 0b97430 --final-tag 1.0.0`

Reviewer 2: ```suggestion
`git changelog --start-commit {{0b97430}} --final-tag {{1.0.0}}`
```

> `git changelog CHANGELOG.md`

Reviewer 2: ```suggestion
`git changelog {{CHANGELOG.md}}`
```
